process/economics/rewardsPerBlock.go

```diff
@@ -0,0 +1,233 @@
+package economics
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/block"
+	"github.com/ElrondNetwork/elrond-go/dataRetriever"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+)
+
+type rewardsPerBlock struct {
+	marshalizer      marshal.Marshalizer
+	store            dataRetriever.StorageService
+	shardCoordinator sharding.Coordinator
+	nodesCoordinator sharding.NodesCoordinator
+	rewardsHandler   process.RewardsHandler
+}
+
+// ArgsNewEpochEconomics
+type ArgsNewEpochEconomics struct {
+	Marshalizer      marshal.Marshalizer
+	Store            dataRetriever.StorageService
+	ShardCoordinator sharding.Coordinator
+	NodesCoordinator sharding.NodesCoordinator
+	RewardsHandler   process.RewardsHandler
+}
+
+// NewEndOfEpochEconomicsDataCreator creates a new end of epoch economics data creator object
+func NewEndOfEpochEconomicsDataCreator(args ArgsNewEpochEconomics) (*rewardsPerBlock, error) {
+	if check.IfNil(args.Marshalizer) {
+		return nil, process.ErrNilMarshalizer
+	}
+	if check.IfNil(args.Store) {
+		return nil, process.ErrNilStore
+	}
+	if check.IfNil(args.ShardCoordinator) {
+		return nil, process.ErrNilShardCoordinator
+	}
+	if check.IfNil(args.NodesCoordinator) {
+		return nil, process.ErrNilNodesCoordinator
+	}
+	if check.IfNil(args.RewardsHandler) {
+		return nil, process.ErrNilRewardsHandler
+	}
+
+	r := &rewardsPerBlock{
+		marshalizer:      args.Marshalizer,
+		store:            args.Store,
+		shardCoordinator: args.ShardCoordinator,
+		nodesCoordinator: args.NodesCoordinator,
+		rewardsHandler:   args.RewardsHandler,
+	}
+	return r, nil
+}
+
+// ComputeRewardsPerBlock calculates the rewards per block value for the current epoch
+func (r *rewardsPerBlock) ComputeEndOfEpochEconomics(
+	metaBlock *block.MetaBlock,
+) (*block.Economics, error) {
+	if check.IfNil(metaBlock) {
+		return nil, process.ErrNilHeaderHandler
+	}
+	if metaBlock.AccumulatedFeesInEpoch == nil {
+		return nil, process.ErrNilTotalAccumulatedFeeInEpoch
+	}
+	if !metaBlock.IsStartOfEpochBlock() || metaBlock.Epoch < 1 {
+		return nil, process.ErrNotEpochStartBlock
+	}
+
+	noncePerShardPrevEpoch, prevEpochStart, err := r.startNoncePerShardFromPreviousEpochStart(metaBlock.Epoch - 1)
```

Comment:
suggestion to add an "s" in the naming as to suggest that this a collection rather than a value:
noncePerShardPrevEpoch -> nonce*s*PerShardPrevEpoch
Valid also for noncePerShardCurrEpoch

Reply:
done.